src/lib/modules/debugger/debugger_manager.ts

```diff
@@ -0,0 +1,103 @@
+const RemixDebug = require("remix-debug-debugtest");
+const CmdLine = RemixDebug.CmdLine;
+const async = require("async");
+
+export default class DebuggerManager {
+  private nodeUrl: string;
+  private outputJson: any;
+  private inputJson: any;
+  private isDebugging: boolean;
+
+  constructor(nodeUrl: string) {
+    this.nodeUrl = nodeUrl;
+    this.outputJson = {};
+    this.inputJson = {};
+    this.isDebugging = false;
+  }
+
+  public setInputJson(inputJson: any) {
+    this.inputJson = inputJson;
+  }
+
+  public setOutputJson(outputJson: any) {
+    this.outputJson = outputJson;
+  }
+
+  public createDebuggerSession(txHash: string, filename: string, cb: any) {
+    return this.debug(txHash, filename, cb);
+  }
+
+  private debug(txHash: string, filename: string, cb: any) {
+    const cmdLine = new CmdLine();
+    cmdLine.connect("http", this.nodeUrl);
+    cmdLine.loadCompilationData(this.inputJson, this.outputJson);
+
+    cmdLine.initDebugger(() => {
+      this.isDebugging = true;
+
+      cmdLine.startDebug(txHash, filename, () => {
+        if (cb) {
+          cmdLine.triggerSourceUpdate();
+          cb();
+        }
+      });
+    });
+    return cmdLine;
+  }
+
+  public getLastLine(txHash: string, filename: string, outputCb: any) {
+    const self = this;
+    let cmdLine = new CmdLine();
+
+    async.waterfall([
+      function initDebugger(next: any) {
+        cmdLine = new CmdLine();
+        cmdLine.connect("http", self.nodeUrl);
+        cmdLine.loadCompilationData(self.inputJson, self.outputJson);
+        cmdLine.initDebugger(() => {
+          // self.isDebugging = true
+          next();
+        });
+      },
+      function startDebug(next: any) {
+        const debuggerData: any = {};
+        cmdLine.events.on("locals", (data: any) => {
+          debuggerData.locals = self.simplifyDebuggerVars(data);
+        });
+
+        cmdLine.events.on("globals", (data: any) => {
+          debuggerData.contract = self.simplifyDebuggerVars(data);
+        });
+
+        cmdLine.startDebug(txHash, filename, () => {
+          cmdLine.events.on("source", () => {
+            const lines: string[] = cmdLine.getSource();
+            // TODO: this is a bit of a hack
+            const line: string = lines.filter((x: string) => x.indexOf("=>") === 0)[0];
```

Comment:
Since `lines` is already `string[]` I think there's no need to annotate `x` here. TS is smart enough to figure this out.

Reply:
strange, I would swear the linter was forcing this. it seems to work though. removing...